zulip/integrations/git/post-receive

```diff
@@ -37,15 +38,21 @@ def git_repository_name() -> Text:
         return os.path.basename(os.path.dirname(os.getcwd()))
 
 def git_commit_range(oldrev: str, newrev: str) -> str:
+    remote_repo_cmd = ["git", "config", "--get", "remote.origin.url"]
+    remote_repo_url = subprocess.check_output(remote_repo_cmd).strip().decode("utf-8")
+
     log_cmd = ["git", "log", "--reverse",
-               "--pretty=%aE %H %s", "%s..%s" % (oldrev, newrev)]
+               "--pretty=-[AUTHOR]=%aN-[COMMIT]=%H-[SUBJECT]=%s", "%s..%s" % (oldrev, newrev)]
```

Comment:
It’d be more robust and simpler to separate with newlines (`--pretty=%aN%n%H%n%h%n%s`). This can be parsed without any fanfare:

```python
it = subprocess.check_output(log_cmd, universal_newlines=True).splitlines()
for author_name in it:
    commit_hash = next(it)
    commit_short_hash = next(it)
    subject = next(it)
    commits += git.COMMIT_ROW_TEMPLATE.format(
        author_name=author_name,
        commit_short_hash=commit_short_hash,
        subject=subject,
        commit_url="{}/commit/{}".format(remote_repo_url, commit_hash),
    )
```